src/diepvries/template_sql/satellite_dml.sql

```diff
@@ -1,64 +1,65 @@
 MERGE INTO {target_schema}.{target_table} AS satellite
   USING (
-    WITH
-      filtered_staging AS (
-        SELECT DISTINCT
-          staging.{hashkey_field},
-          staging.{staging_hashdiff_field},
-          staging.{record_start_timestamp},
-          staging.{record_source}
-          {staging_descriptive_fields}
-        FROM {staging_schema}.{staging_table} AS staging
-          CROSS JOIN (
-                       SELECT
-                         MAX({record_start_timestamp}) AS max_r_timestamp
-                       FROM {target_schema}.{target_table}
-                     ) AS max_satellite_timestamp
-        WHERE staging.{record_start_timestamp} >= COALESCE(max_satellite_timestamp.max_r_timestamp, '1970-01-01 00:00:00')
-      ),
-      staging_satellite_affected_records AS (
-        /* Records that will be inserted (don't exist in target table or exist
-          in the target table but the hashdiff changed). As the r_timestamp is fetched
-          from the staging table, these records will always be included in the
-          WHEN NOT MATCHED condition of the MERGE command. */
-        SELECT
-          staging.{hashkey_field},
-          staging.{staging_hashdiff_field},
-          staging.{record_start_timestamp},
-          staging.{record_source}
-          {staging_descriptive_fields}
-        FROM filtered_staging AS staging
-          LEFT OUTER JOIN {target_schema}.{target_table} AS satellite
-                          ON (staging.{hashkey_field} = satellite.{hashkey_field}
+        WITH
+          filtered_staging AS (
+          SELECT DISTINCT
+            staging.{hashkey_field},
+            staging.{staging_hashdiff_field},
+            staging.{record_start_timestamp},
+            staging.{record_source}
+            {staging_descriptive_fields}
+          FROM {staging_schema}.{staging_table} AS staging
+          WHERE NOT EXISTS (
+                           SELECT
+                             1
+                           FROM {target_schema}.{target_table} AS satellite
+                           WHERE staging.{hashkey_field} = satellite.{hashkey_field}
+                             AND satellite.r_timestamp >= staging.r_timestamp
+                           )
```

Comment:
Hum good point. before we were doing `WHERE staging.{record_start_timestamp} >= COALESCE(max_satellite_timestamp.max_r_timestamp, '1970-01-01 00:00:00')`, so the other way around. 

We should do `satellite.r_timestamp > staging.r_timestamp`.

Although, I wonder if this is really the correct behaviour. I'd argue that the previous one could be incorrect. Inserting a record with the exact same timestamp will also mess up the calculation of the `r_timestamp_end`. It is SUPER unlikely to happen and almost impossible to simulate, but I think the new behavior is more correct! No?

Reply:
Ah I think I remember. The difference here is that in the first union all query we're fetching the fields from the `staging.<field_name>` and the second we fetch the fields from the `satellite.<field_name>`.

So in the first union all we want either the new version  of existing records (source from stg, different hashdiff) or the first version of records that never existed!

Whereas in the next one we want to fetch the existing version from the sateliite with new versions on the staging table.

The key here is that the `staging.{record_start_timestamp}` will always be different than the `satellite.{record_start_timestamp}` so these will always be different versions and both should be considered in the query as we do a merge by primary key (hashkey + r_timestamp or driving key + r_timestamp)

Does this make sense?

Reply:
Yeah, I just wanted to point out that this was a small change. I think the new behaviour is simpler, specially because of these messed up calculation of `r_timestamp_end` that could happen. So I agree, let's do the change